data/transactions/logic/assembler.go

```diff
@@ -1265,13 +1286,99 @@ func (ops *OpStream) checkStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+const subRets int = -1
+const nowhere int = -2
+const exiting int = -3
+const erroring int = -4
+
+//A BasicBlock is a structure through which control can only flow in through start and out through end
+type BasicBlock struct {
+	startIndex       int         //index into ops.pending.Bytes()
+	endIndex         int         //Note this is the beginning of the last op in the block, somewhat unnecessary but we'll keep it for now
+	expectedStackTop []StackType //What the block requires to be on top of the stack in order to complete
+	stackDelta       []StackType //What the block net pushes to the stack
+	sourceErrors     []int       //Sourcelines for each entry in expectedStackTop which is useful for listing sourcelines for errors
```

Comment:
You mean the line that "consumes" the value?

Reply:
Ya, I wasn't sure of a good name for this one or really a good way to explain it.